### PR TITLE
Check administrator only when security is enabled

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

This PR aims to improve out-of-box UX.

Currently, administrators' checking is enforced even when security is disabled, and we are going to deliver the preview of Web UI in 1.8, it may confuse users who use the default configuration to launch Kyuubi sever and explore Kyuubi WebUI (because the user "anonymous" can not visit all users' sessions/engines without being configured as an administrator).

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request

### _Was this patch authored or co-authored using generative AI tooling?_

No

### CODE REVIEW

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiRestFrontendService.scala [hunk 1]
```diff
@@ -70,6 +71,17 @@ class KyuubiRestFrontendService(override val serverable: Serverable)
 
   private lazy val port: Int = conf.get(FRONTEND_REST_BIND_PORT)
 
+  private lazy val isSecurityEnabled = {
+    val authTypes = conf.get(AUTHENTICATION_METHOD).map(AuthTypes.withName)
+    !authTypes.contains(NONE) && authTypes != Set(NOSASL)
```
Comment (lightning-L) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiRestFrontendService.scala:76]: Just a trivial question, is it possible to reuse the variables defined in `KyuubiAuthenticationFactory`.
```
  private val authTypes = conf.get(AUTHENTICATION_METHOD).map(AuthTypes.withName)
  private val none = authTypes.contains(NONE)
  private val noSasl = authTypes == Set(NOSASL)
```

**Review by lightning-L — APPROVED** [review]
LGTM

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiRestFrontendService.scala (hunk 1 above)
Comment (pan3793) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiRestFrontendService.scala:76]: Thanks for pointing out it, actually, I found bunches of duplicated code in THRIFT-HTTP and RESTful protocols, I'm fixing the tests and reducing the code duplications now.

Comment (pan3793) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiRestFrontendService.scala]: FYI: I refactored code to reuse `getValidPasswordAuthMethod` as you suggested

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/http/authentication/AuthenticationHandler.scala [hunk 2]
```diff
@@ -103,23 +101,10 @@ trait AuthenticationHandler {
...
-
-  /**
-   * HTTP header used by the SPNEGO server endpoint during an authentication sequence.
-   */
-  final val WWW_AUTHENTICATE: String = HttpConstants.WWW_AUTHENTICATE_HEADER
-
-  /**
-   * HTTP header used by the client endpoint during an authentication sequence.
-   */
-  final val AUTHORIZATION_HEADER = "Authorization"
```
Comment (pan3793) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/http/authentication/AuthenticationHandler.scala:124]: moved such definitions to `HttpAuthUtils`

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiRestFrontendService.scala [hunk 3]
```diff
@@ -240,7 +252,7 @@ class KyuubiRestFrontendService(override val serverable: Serverable)
       realUser
     } else {
       sessionConf.get(KyuubiAuthenticationFactory.HS2_PROXY_USER).map { proxyUser =>
-        if (!getConf.get(KyuubiConf.SERVER_ADMINISTRATORS).contains(realUser)) {
+        if (!isAdministrator(realUser)) {
```
Comment (pan3793) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiRestFrontendService.scala:255]: the behavior here has changed, previously, `Utils.currentUser` was not counted, which is inconsistent with checking administrator logic in `AdminResource`

**Comment by zhouyifan279** [comment]
LGTM

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiRestFrontendService.scala (hunk 1 above)
Comment (pan3793) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiRestFrontendService.scala]: I mis-interpreted the suggestion, you suggested `AuthTypes` but I used `AuthMethods`, which cause `IllegalArgumentException("No valid Password Auth detected")` when configuring `kyuubi.authentication=KERBEROS`, will fix it soon.

**Pull request closed without merging**